_posts/2021-01-09-Julia-1.6-what-has-changed-since-1.0.md

```diff
@@ -0,0 +1,639 @@
+This is going to help prevent people knew the language from including compilation time in their benchmarks.
+It is still better to use [BenchmarkTool.jl's](https://github.com/JuliaCI/BenchmarkTools.jl) `@btime`, since that does multiple samples.
+But now that too can report time spend compilation.
+It's also useful for identifying if certain functions are taking ages to compile.
+Which I guess is its theoretical main point, but I think preventing people benchmarking wrong is going to come up way more often.
+
+The experimental [`Base.@locals`](https://github.com/JuliaLang/julia/issues/29733
+) was added which returns a dictionary of local variables.
+That one surprised me, I though being able to access a dict of local variables would get in the way of the optimizer, since it would prevent it from being able to optimize variables that are used for intermediate values away entirely.
+But the compiler folk know better than I do.
+
+[`splitpath`](https://github.com/JuliaLang/julia/issues/28156) as added, its the opposite of `joinpath`.
+Kind of silly we didn't have that, and had been being me at least since 0.6.
+
+On things that had been bugging me, I had wanted [`eachslice` and it's special cases: `eachrow` & `eachcol`](https://github.com/JuliaLang/julia/issues/29749) since julia 0.3 when I first started using it.
+These are super handy when you want to e.g. iterate through vectors of the rows of a matrix.
+
+`redirect_stderr` and `redirect_stdout` now work with `devnull`.
+So one can run some suppressing output easily as follows:
+```julia
+julia> redirect_stdout(devnull) do
+       println("You won't see this")
+       end
+```
+This is just handy, doing it without this is seriously annoying.
+
+`readdir` now accepts a `join=true|false` keyword argument so that it returns paths with the parent dir.
+This is good, almost every time I use `readdir` I used it as:
+`joinpath.(x, readdir(x))`.
+It is slightly cleaner (and faster) to be able to do `readdir(x; join=true)`.
+I think for Julia 2.0 we should consider making it the default.
+Also added was a `sort` argument, which I don't see the point of so much, since `sort(readdir(x))` seems cleaner than `readdir(x; sort=true)`; and because I rarely rely on processing files in order.
+
+
+`ccall` is now available as a macro `@ccall` which lets you specify 
+There was a [short juliacon talk about this](https://www.youtube.com/watch?v=wofq1DdXM3s)
+So now one can do `@ccall(sqrt(4.0::Cdouble)::Cdouble)`
+rather than `ccall(:sqrt, Cdouble, (Cdouble,), 4.0,)`
+This is what `ccall` always should have been.
+In julia 1.7, [`invoke` is getting the same treatment.](https://github.com/JuliaLang/julia/pull/38438).
+
+
+### More "Why didn't it always work that way" than I can count
+Since 1.0's release there have been so many small improvements to functions that I didn't even know happened, because I assumed they always worked that way.
```

Comment:
```suggestion
Since 1.0's release, there have been so many small improvements to functions that I didn't even know happened, because I assumed they always worked that way.
```